addons/website_sale_comparison/static/src/js/website_sale_comparison.js

```diff
@@ -113,15 +115,19 @@ var ProductComparison = Widget.extend({
         }
     },
     add_new_products:function(product_id){
+        this.guard.exec(this._add_new_products.bind(this, product_id));
+    },
+    _add_new_products: function (product_id) {
         var self = this;
         $('.o_product_feature_panel').show();
         if (!_.contains(self.comparelist_product_ids, product_id)) {
             self.comparelist_product_ids.push(product_id);
             if(_.has(self.product_data, product_id)){
                 self.update_content([product_id], false);
             } else {
-                self.load_products([product_id]).then(function(){
+                return self.load_products([product_id]).then(function () {
                     self.update_content([product_id], false);
+                    self.update_cookie();
```

Comment:
Would have been better to share update_content followed by update_cookie to not duplicate the code but I guess as it is a fix for 11.0, it is better to avoid conflicts.